Cleaning reconst scripts part2

EmmaRenauld opened the pull request:
# Quick description

Part 2 in my meticulous verification of reconst scripts. Read the 9 next scripts in our excel list!

Almost all already correct! Changes a few small code style along the way. The only real changes:

- In fodf_max_in_ventricles: 
       - Removed the "useful for 2D data" in the help. We don't manage the 2D. We manage 3D data with one dimension set to 1, but we don't add the 3rd dimension ourselves.
      - Changed the method it calls so that we don't send `args`. Will be easier to create a script.
- scil_freewater_priors: only calls noddi_priors, but had a coverage of 0%. Added a test.

That's it!
...

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Verifications

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

karanphil commented on scripts/scil_fodf_max_in_ventricles.py:
```diff
@@ -40,25 +40,23 @@ def _build_arg_parser():
-                        ' are considered for evaluation, [%(default)s]).')
-    p.add_argument('--md_t', dest='md_threshold',
-                   type=float, default='0.003',
+    p.add_argument('--fa_threshold', type=float, default='0.1',
```
I would argue that `--fa_thr` is enough and shorter to write.

karanphil replied:
Same for MD

EmmaRenauld replied:
I can still write it -fa_t when you call your script, it will work. But I think it's clearer with the whole name. No?

karanphil replied:
Alright, I was just to be consistent with other scripts, but at least I prefer --fa_threshold over --fa_t.

karanphil commented on scripts/scil_freewater_priors.py:
```diff
@@ -2,20 +2,11 @@
 """
-Compute the axial (para_diff) and mean (iso_diff) diffusivity priors for
-freewater.
+Synonym for scil_NODDI_priors.py
```
I don't know why we would reduce the explanation here. I would add your sentence after what was previously here.

EmmaRenauld replied:
Because it does not show anywhere. If you do `scil_freewater_priors -h`, it prints the expanation of scil_NODDI_priors.

karanphil replied:
Ah I see!

karanphil commented on scripts/scil_frf_ssst.py:
```diff
@@ -53,24 +53,20 @@ def _build_arg_parser():
                         'function.')
-    p.add_argument('--fa', dest='fa_thresh',
-                   default=0.7, type=float,
+    p.add_argument('--fa_thresh', default=0.7, type=float,
```
Again, I would vote for `--fa_thr`, and `--min_fa_thr` perhaps.

karanphil reviewed (commented):
Just a few minor comments! Good job.

pep8speaks commented:
Hello @EmmaRenauld, Thank you for updating !



















* In the file [`scripts/scil_fodf_memsmt.py`](https://github.com/scilus/scilpy/blob/93ce6474be3c598dc52892f92a1b3b6bd37334be/scripts/scil_fodf_memsmt.py):

> [Line 57:80](https://github.com/scilus/scilpy/blob/93ce6474be3c598dc52892f92a1b3b6bd37334be/scripts/scil_fodf_memsmt.py#L57): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)



* In the file [`scripts/scil_fodf_msmt.py`](https://github.com/scilus/scilpy/blob/93ce6474be3c598dc52892f92a1b3b6bd37334be/scripts/scil_fodf_msmt.py):

> [Line 40:80](https://github.com/scilus/scilpy/blob/93ce6474be3c598dc52892f92a1b3b6bd37334be/scripts/scil_fodf_msmt.py#L40): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)

EmmaRenauld commented:
Finally I added a bit more changes!

- Combined common sections in fodf_memst, fodf_msmt.
- Added small changes in bingham_metrics, which I just found was not in our excel sheet.
- Encapsulated things in frf_set_diffusivities.

Overall, I thus have mainly small changes, and 3 new functions (created without changing any line of code!). I was not always sure where to put them though.

arnaudbore reviewed (approved):
Some quick comments. LGTM

EmmaRenauld commented on scilpy/reconst/sh.py:
```diff
@@ -17,6 +17,26 @@
 from scilpy.dwi.operations import compute_dwi_attenuation
 
 
+def verify_data_vs_sh_order(data, sh_order):
```
I just checked really fast but I'm not sure where to check or what to search for. @karanphil ?

karanphil replied:
I don't think they have this.

karanphil commented on scilpy/reconst/fodf.py:
```diff
@@ -174,3 +184,58 @@ def fit_from_model(model, data, mask=None, nbr_processes=None):
+                         'Invalid or deprecated FRF format')
+    if not csf_frf.shape[1] == 4:
+        raise ValueError('CSF frf file did not contain 4 elements. '
+                         'Invalid or deprecated FRF format')
```
This should return the frfs.

arnaudbore commented on scilpy/reconst/fodf.py:
```diff
@@ -174,3 +188,71 @@ def fit_from_model(model, data, mask=None, nbr_processes=None):
+    ----------
+    wm_frf: np.ndarray
+    gm_frf: np.ndarray
+    csf_frf: np.ndarray
```
Missing returns

arnaudbore reviewed (commented):
Quick fix and then GTG

arnaudbore reviewed (approved):
Seems GTG !
